### ISSUE TITLE
RFM needs a pagination

### ISSUE THREAD

**Issue by qsad** [trippo/ResponsiveFilemanager#383]
I've tried to use RFM for a current website with huge uploads folder (more than 30k files) which I need to be able to explore and use via RFM.
But it had just frozen and finally crashed my browser tab. I'm sure it's because my uploads folder is too huge.
I think just simple pagination or infinite scroll would fix this problem. Cheers!

**Comment by trippo** [comment]
You are right but for do that i need to convert the list in a db, actually simply read the file system

**Comment by qsad** [comment]
Thanks, could you please say me is it possible to do it by myself or not?

**Comment by trippo** [comment]
I think yes, on upload you can save in db and on dialog.php you can read db instead filesystem

**Comment by Hibbelharry** [comment]
This is really an issue. We, for example, do have an image upload folder in our project, which easily grows to around 10k files in it. When we put all references in our html, we get docsizes of around 55MB, which take a while to download. After that, js parsing takes another while, too.

I don't think you should build a db to index files. DBs of indexed files are always behind the current state of a folder, so thats not what I would do. I think getting a file list should be done by creating an API endpoint inside of RFM, returning a json with a filelist. If that endpoint gets some parameters, i.e. endpoint.php?numberofiles=100&start=300&sort=alphabet&order=descending, you could get nicely short parts of the whole list, and would move parts of processing out of slow JS Code.

**Comment by mpryvkin** [comment]
To solve the problem with large number of files in the folder I added support for lazy loading of files with either "Load more" button or auto-loading of more files when user scrolls all the way down. I may update forked repo and will send PR if my client allows me.

**Comment by navotera** [comment]
So what the current status of this features ? Seem like what @mpryvkin  not merged to the repo yet @trippo

**Comment by mpryvkin** [comment]
@navotera See #485 where @trippo decided not to merge the pull request.

**Comment by navotera** [comment]
We need this feature can you add download link of this RFM along with your
feature somewhere?

**Comment by mpryvkin** [comment]
@navotera Please see [this branch](https://github.com/mpryvkin/ResponsiveFilemanager/tree/feature-load-more).

**Comment by navotera** [comment]
Thank you so much @mpryvkin 

Hope you have a great day 👍
